### DISCUSSION TITLE
Bug: Filtering by ColumnTrait.eq(None) does not filter by NULL

### DISCUSSION THREAD

**Discussion by Trivernis** [SeaQL/sea-orm#268] [Q&A] [answered]
## Summary

When using the expression `mytable::Column::MyColumn.eq(None)` the resulting sql statement is `...column = ?`. This expression filters by value and therefore does not filter for `NULL` since `NULL` is not a value in the sql syntax.

## Expected Result

The resulting query should filter by `IS NULL` instead. This type of expression can currently only be constructed when using the sea_query expression builder. 

## System information

Database: Sqlite3
Crate Version: 0.2.4

**Comment by tyt2y3** [comment, marked as answer]
- you should use [is_null](https://docs.rs/sea-query/0.18.0/sea_query/expr/struct.Expr.html#method.is_null) instead of eq None

Not sure why it was missing from ColumnTrait though, we will add it.

Meanwhile you can construct the expression yourself as in 

```rust
use sea_query::Expr;

.filter(Expr::col(fruit::Column::CakeId).is_null())
```

**Reply by tyt2y3** [reply]
[0.3.1](https://github.com/SeaQL/sea-orm/releases/tag/0.3.1)
